From c4f216359258206fd2b997c5dda1f50892e30301 Mon Sep 17 00:00:00 2001
From: Mark Brown <broonie@sirena.org.uk>
Date: Thu, 3 May 2012 10:27:58 +0100
Subject: [PATCH] Restore gzflags symbol for ABI compatibility

zlib 1.2.7 no longer uses the gzflags symbol but since it was previously
an exported symbol we should really retain it in order to preserve
binary compatibility in case anything linked with it; nothing should
have been doing so but the cost of retaining the symbol is very low so
just do it.
---
 zlib.map |    1 +
 zutil.c  |    6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/zlib.map b/zlib.map
index 771f420..54fa553 100644
--- a/zlib.map
+++ b/zlib.map
@@ -73,6 +73,7 @@ ZLIB_1.2.5.1 {
 
 ZLIB_1.2.5.2 {
     deflateResetKeep;
+    gzflags;
     gzgetc_;
     inflateResetKeep;
 } ZLIB_1.2.5.1;
diff --git a/zutil.c b/zutil.c
index 65e0d3b..ddece96 100644
--- a/zutil.c
+++ b/zutil.c
@@ -321,4 +321,10 @@ void ZLIB_INTERNAL zcfree (opaque, ptr)
 
 #endif /* MY_ZCALLOC */
 
+/* Dummy function returned for ABI compatibility */
+unsigned long ZEXPORT gzflags (void)
+{
+    return 0;
+}
+
 #endif /* !Z_SOLO */
-- 
1.7.10

